### PR TITLE
chore: add eslint and markdownlint

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description

Now that we have Vue components and more javascript code in Dialtone we are adding ESLint.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [ ] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

File: docs/.vuepress/theme/components/Navbar.vue [hunk 1]
```diff
@@ -16,112 +33,161 @@
...
+          <svg
+            aria-hidden="true"
+            class="d-svg d-svg--system d-svg--Github d-svg--size20"
+            height="24"
+            viewBox="0 0 18 18"
+            width="24"
+          >
             <path
-              d="M9 1a8 8 0 00-2.53 15.59c.4.07.55-.17.55-.38l-.01-1.49c-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82a7.42 7.42 0 014 0c1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48l-.01 2.2c0 .21.15.46.55.38A8.01 8.01 0 009 1z"></path>
+              d="M9 1a8 8 0 00-2.53 15.59c.4.07.55-.17.55-.38l-.01-1.49c-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.
```
Comment (josedialpad) [inline docs/.vuepress/theme/components/Navbar.vue]: These icons are broken
2022-07-05 at 11 59 59@2x

Comment (braddialpad) [reply docs/.vuepress/theme/components/Navbar.vue]: Bit surprised the linter did this here. This isn't really a valid way to split a string?

Comment (juliodialpad) [reply docs/.vuepress/theme/components/Navbar.vue]: good catch! thanks

File: .eslintrc.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ module.exports = {
     'max-len': ['error', {
       code: 120,
       tabWidth: 2,
+      ignorePattern: 'd="([\\s\\S]*?)"',
```
Comment (braddialpad) [inline .eslintrc.js]: Maybe `'<path d="([\\s\\S]*?)"'` just to be safe?

Comment (braddialpad) [reply .eslintrc.js]: I guess it's possible for d to not be first, so maybe something like

`'(<path)[\s\S\n]*?d="([\\s\\S]*?)"'`

Comment (juliodialpad) [reply .eslintrc.js]: Or I might just disable eslint on those 2 lines, seems to be much simpler, what do you think?

File: docs/.vuepress/theme/components/Navbar.vue (hunk 1 above)
Comment (juliodialpad) [reply docs/.vuepress/theme/components/Navbar.vue]: > Bit surprised the linter did this here. This isn't really a valid way to split a string?

I did it manually, but seems like linter might be removing spaces and that's why the icons end up broken.

File: .eslintrc.js (hunk 2 above)
Comment (juliodialpad) [reply .eslintrc.js]: or move those to svg icons instead of having them inline

Comment (braddialpad) [reply .eslintrc.js]: just making them svg icons might be the way to go.

**Review by braddialpad — APPROVED** [review]
Thanks Julio, this is awesome!

**Pull request merged**